### DISCUSSION TITLE
Performance Issues (Sorting, etc.)

### DISCUSSION THREAD

**Discussion by nickheyer** [sagiegurari/duckscript#469] [General]
I recently did some serious stress tests on the duck script language, embedded in an application I'm working on.

I may be missing something, but I implemented a standard bubble sort and swap sort in a test script. My results were somewhat surprising.

Test:

Given a text file, containing a pair of numbers on each line (1000 lines) in the following format "1244      5432", parse the text into two unsorted arrays (left and right), then sort each array. 

Result:

Bubble Sort... 22s
Left - Sorted in 11s
Right - Sorted in 11s

Swap Sort... 12s 30ms
Left - Sorted in 12s
Right - Fn returned after 30ms, returned array unsorted

I didn't think this language would be incredibly performant, but these results were surprising. The manual sort code was pretty lengthy given that there isn't any sort functions in sdk. Can share if it would help.

**Comment by sagiegurari** [comment]
maybe worth having a sort function in the SDK. if you want to implement one.
however, the lang wasn't meant to be fast (not that i want it to be slow :) ) but to be a very simple shell like lang that people do'nt have to learn or invest in order to do simple things and go on with their lives. main usage is cargo make scripting of course.
those are usually very small scripts.
